File: day1/day1.py
import re

swap = {
    "one": "o1e",
    "two": "t2o",
    "three": "t3e",
    "four": "f4r",
    "five": "f5e",
    "six": "s6x",
    "seven": "s7n",
    "eight": "e8t",
    "nine": "n9e",
}


def test():
    assert extract_numbers("8ee4wd2") == "842"
    assert convert_numbers("eightwo") == "e8t2o"
    assert score("853") == 83
    assert score("5") == 55


def extract_numbers(s: str) -> str:
    return "".join([n for n in re.findall(r"(\d)", s)])


def convert_numbers(s: str) -> str:
    for key in swap.keys():
        s = s.replace(key, swap[key])
    return s


def score(s: str) -> int:
    try:
        return int(s[0] + s[-1])
    except:
        raise ValueError


def main():
    test()
    s1 = 0
    s2 = 0
    with open("input.1") as ifile:
        for line in ifile:
            s1 += score(extract_numbers(line))
            s2 += score(extract_numbers(convert_numbers(line)))
    print(f"s1:\t{s1}")
    print(f"s2:\t{s2}")


if __name__ == "__main__":
    main()

File: day1/go/day1.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"regexp"
	"strconv"
	"strings"
)

func convertNums(l string) string {
	r := strings.NewReplacer("one", "1e", "two", "2o", "three", "3e", "four", "4r", "five", "5e", "six", "6x", "seven", "7n", "eight", "8t", "nine", "9e")
	for {
		if l == r.Replace(l) {
			break
		}
		l = r.Replace(l)
	}
	return l
}

func cleanNums(l string) string {
	re := regexp.MustCompile(`[0-9]+`)
	x := re.FindAllString(l, -1)
	return strings.Join(x, "")
}

func score(l string) int {
	var s []string
	s = append(s, string(l[0]))
	s = append(s, string(l[len(l)-1]))
	l = strings.Join(s, "")
	i, err := strconv.Atoi(l)
	if err != nil {
		panic(err)
	}
	return i
}

func main() {
	t1 := 0
	t2 := 0
	f, err := os.Open("../input.1")
	if err != nil {
		panic(err)
	}
	defer f.Close()
	scanner := bufio.NewScanner(f)
	for scanner.Scan() {
		l := scanner.Text()
		fmt.Println(l)
		t1 += score(cleanNums(l))
		t2 += score(cleanNums(convertNums(l)))
	}
	fmt.Printf("P1:%d\tP2:%d\n", t1, t2)
	if scanner.Err() != nil {
		panic(scanner.Err())
	}
}
